pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStore.java

```diff
@@ -87,94 +87,109 @@ public MLTransactionMetadataStore(TransactionCoordinatorID tcID,
         this.abortedTransactionCount = new LongAdder();
         this.transactionTimeoutCount = new LongAdder();
         this.appendLogCount = new LongAdder();
+    }
 
+    public CompletableFuture<TransactionMetadataStore> init(TransactionRecoverTracker recoverTracker) {
+        CompletableFuture<TransactionMetadataStore> completableFuture = new CompletableFuture<>();
         if (!changeToInitializingState()) {
             log.error("Managed ledger transaction metadata store change state error when init it");
-            return;
-        }
-        new Thread(() -> transactionLog.replayAsync(new TransactionLogReplayCallback() {
-
-            @Override
-            public void replayComplete() {
-                recoverTracker.appendOpenTransactionToTimeoutTracker();
-                if (!changeToReadyState()) {
-                    log.error("Managed ledger transaction metadata store change state error when replay complete");
-                } else {
-                    recoverTracker.handleCommittingAndAbortingTransaction();
-                    timeoutTracker.start();
+            completableFuture
+                    .completeExceptionally(new TransactionCoordinatorClientException
+                    .CoordinatorNotFoundException("transaction metadata store with tcId "
+                            + tcID.toString() + " change state to Initializing error when init it"));
+        } else {
+            new Thread(() -> transactionLog.replayAsync(new TransactionLogReplayCallback() {
```

Comment:
now that we are putting our hands in this code.
can we remove this "new Thread" ?
we should use one ThreadPool that is already available and not start a new Thread out of any resource management pool.

Reply:
+1 we already introduced the transaction thread pool, @liangyepianzhou is there any problem if using the transaction thread pool?

Reply:
Are you referring to this: 
```java
 if (config.isTransactionCoordinatorEnabled()) {
            this.transactionExecutorProvider = new ExecutorProvider(this.getConfiguration()
                    .getNumTransactionReplayThreadPoolSize(), "pulsar-transaction-executor");
        } else {
            this.transactionExecutorProvider = null;
        }
```

Reply:
1. We have used this thread pool in TB. We don't want TB and TC to use the same thread pool at the same time.

2. The optimization of TC is temporarily slowed down, and the optimization will be carried out after the correction of APi is completed.

Reply:
I have change it to a single-threaded pools at https://github.com/apache/pulsar/pull/14238